src/Service/MailTemplateHelper.php

```diff
@@ -498,6 +499,25 @@ private function flatten(array $value): array
         return array_combine($header, $row);
     }
 
+    public function computeMergeFields(MailTemplate $template)
```

Comment:
Rename to `computeCustomFields` or `getCustomFieldNames`. It may even make sende to move it to the Template class since it doesn't depend on services and any other fancy stuff.

The current function body can be reduced to a one-liner (split over multiple lines for readability):

```php
return array_filter(
  array_map('trim'
    explode(
      PHP_EOL,
      $template->getCustomFields()
    )
  )
);
```

But it may be better to return a list of object for future extensions:

```php
return [
  [
    'name' => 'my_field',
  ],
];
```

as this will make it easier to add labels (and maybe even type (textarea, text, number)) in the future.